### PR TITLE
Feat/1/Setup Repository to Use Nodejs

### LINKED ISSUE #1: Setup Repository to Use Nodejs
**Description**
Set up the user service repository using the nodejs with clean code architecture.

**How To**
- Init the nodejs codebase.
- Create the DDD layer in the app including presentation layer, domain layer, infrastructure layer. Reference: https://github.com/ricomonster/node-ddd
- Create an endpoint to check the service's health to check the database is running properly.
    - Url is`/healthcheck` with GET method. 
    - Call a DB Ping function to check the db whether running properly or not.
    - Return `200` if the database is running properly.

**Definition of Done**
- Documentation of how to run the project created.
- Service can be running properly.
- Endpoint `/healthcheck` created.

### PR DESCRIPTION
**Pull request by sarahaudina**
close #1

### CODE REVIEW

File: src/common/guard/jwt.guard.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  handleRequest(
+    err: any,
+    user: any,
+    info: any,
+    context: ExecutionContext,
+    status?: any,
+  ) {
+    const request = context.switchToHttp().getRequest();
+    //HACK not recommended though 😅
```
Comment (isfanazha) [inline src/common/guard/jwt.guard.ts]: better to remove this comment 🤣 . maybe can use "//TODO: Need to improve the flow". So using "TODO:" comment, we can easily search which function needs to be improved hehe

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  async verifyUser(dto: LoginDto) {
+    const user = await this.prisma.user.findUnique({
+      where: {
+        email: dto.email,
+      },
+    });
+    if (!user) throw new NotFoundException('user does not exist');
+    const matches = await argon.verify(user.password, dto.password);
+
+    // const hashedInputPassword = await argon.hash(dto.password);
```
Comment (isfanazha) [inline src/auth/auth.service.ts]: if it's unnecessary, just remove it hehe

File: src/auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    const password = await argon.hash(dto.password);
+    try {
+      const user = await this.prisma.user.create({
+        data: {
+          ...dto,
+          password: password,
+        },
+      });
+
+      console.log('user:' + user.password);
```
Comment (isfanazha) [inline src/auth/auth.service.ts]: remove unnecessary console log

File: src/auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        username: user.username,
+        bio: user.bio,
+        image: user.image,
+      };
+
+      return userToReturn;
+    } catch (e) {
+      if (e instanceof PrismaClientKnownRequestError) {
+        console.log('email is taken');
+        if (e.code === 'P2002') throw new BadRequestException('email is taken');
```
Comment (isfanazha) [inline src/auth/auth.service.ts]: what is `P2002` , should we move it to the constant?

**Review by isfanazha — CHANGES_REQUESTED** [review]
minor comment @sarahaudina

File: src/common/guard/jwt.guard.ts (hunk 1 above)
Comment (sarahaudina) [reply src/common/guard/jwt.guard.ts]: 🤣 noted 👌

File: src/auth/auth.service.ts (hunk 2 above)
Comment (sarahaudina) [reply src/auth/auth.service.ts]: sorry i just love hoarding unused code

File: src/auth/auth.service.ts (hunk 4 above)
Comment (sarahaudina) [reply src/auth/auth.service.ts]: its prisma error code for duplicate unique entity

File: src/auth/auth.service.ts (hunk 2 above)
Comment (isfanazha) [reply src/auth/auth.service.ts]: i see haha @sarahaudina

Comment (sarahaudina) [reply src/auth/auth.service.ts]: ✌️

**Comment by sarahaudina** [comment]
Updated with requested changes

**Comment by isfanazha** [comment]
lgtm @sarahaudina

**Review by isfanazha — APPROVED** [review]

**Pull request merged**